docs-md/developer-guide/joins/partition-data.md

```diff
@@ -16,67 +16,70 @@ records based on the joining column. To ensure that records with the same
 of tables, so you can only use a table's primary key as a join column.
 
 Streams, on the other hand, may not have a defined key or may have a key that
-differs from the join column. In these cases, ksqlDB repartitions the stream,
-which implicitly defines a primary key for it. The primary keys for streams
-and tables are of data type `VARCHAR`. 
+differs from the join column. In these cases, ksqlDB internally repartitions
+the stream, which implicitly defines a key for it.
 
-For primary keys to match, they must have the same serialization format. For
-example, you can't join a `VARCHAR` key encoded as JSON with one encoded as
-AVRO.
-
-!!! note
-    ksqlDB requires that keys are encoded as UTF-8 strings.
+ksqlDB requires keys to use the `KAFKA` format. For more information, see
+[Serialization Formats](serialization.md#serialization-formats). If internally
+repartitioning, ksqlDB uses the correct format.
 
 Because you can only use the primary key of a table as a joining column, it's
 important to understand how keys are defined. For both streams and tables, the
-column that represents the primary key has the name `ROWKEY`.
+column that represents the key has the name `ROWKEY`.
 
 When you create a table by using a CREATE TABLE statement, the key of the
 table is the same as that of the records in the underlying Kafka topic.
+You must set the type of the `ROWKEY` column in the
+CREATE TABLE statement to match the key data in the underlying {{ site.ak }} topic.
 
 When you create a table by using a CREATE TABLE AS SELECT statement, the key of
 the resulting table is determined as follows:
 
-- If the FROM clause is a single source, and the source is a stream, the
-  statement must have a GROUP BY clause, where the grouping columns determine
-  the key of the resulting table.
-- If the single source is a table, the key is copied over from the key of the
-  table in the FROM clause. If the FROM clause is a join, the primary key of the
-  resulting table is the joining column, since joins are allowed only on keys.
-- If the statement contains a GROUP BY, the key of the resulting table
-  comprises the grouping columns.
-
-When the primary key consists of multiple columns, like when it's created as
-the result of a GROUP BY clause with multiple grouping columns, you must use
-ROWKEY as the joining column. Even when the primary key consists of a single
-column, we recommend using ROWKEY as the joining column to avoid confusion.
-
-The following example shows a `users` table joined with a `clicks` stream 
+- If the FROM clause contains a stream, the statement must have a GROUP BY clause,
+  and the grouping columns determine the key of the resulting table.
+    - When grouping by a single column or expression, the type of `ROWKEY` in the
+    resulting stream matches the type of the column or expression.
+    - When grouping by multiple columns or expressions, the type of `ROWKEY` in the
+    resulting stream is a `STRING`.
```

Comment:
would be nice to clarify what this `STRING` is (or link to some documentation that clarifies it)

Reply:
Will be done in next PR!